presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -363,6 +373,40 @@ void PrestoServer::run() {
         json infoStateJson = convertNodeState(server->nodeState());
         http::sendOkResponse(downstream, infoStateJson);
       });
+  httpServer_->registerPut(
+    "/v1/info/state",
+    [server = this](
+        proxygen::HTTPMessage* /*message*/,
+        const std::vector<std::unique_ptr<folly::IOBuf>>& body ,
+        proxygen::ResponseHandler* downstream) {
+         std::string requestBody=getRequestBody(body);
+         try {
+            folly::dynamic jsonBody = folly::parseJson(requestBody);
```

Comment:
Let's use `folly::json_patch::try_parse(folly::parseJson(requestBody))`

Reply:
We won't need the try-catch with the `folly::json_patch` usage.

Reply:
Ok.

Reply:
I clarified with @czentgr.
We should put the whole block in a function say `handleGracefulShutdown(body)` similar to other `registerGet` implementations. I agree that this will make it more readable.
```
      "/v1/info/state",
      [server = this](
          proxygen::HTTPMessage* /*message*/,
          const std::vector<std::unique_ptr<folly::IOBuf>>& body,
          proxygen::ResponseHandler* /*downstream*/) {
        server->handleGracefulShutdown(body);
      });
```

Reply:
See discussion above. The body is not JSON string.